index.js

```diff
@@ -98,28 +101,30 @@ CaseSensitivePathsPlugin.prototype.fileExistsWithCase = function(
 
   // Check all filenames in the current dir against current filename to ensure one of them matches.
   // Read from the cache if available, from FS if not.
-  that.getFilenamesInDir(dir, (filenames) => {
-    // If the exact match does not exist, attempt to find the correct filename.
-    if (filenames.indexOf(filename) === -1) {
-      // Fallback value which triggers us to abort.
-      let correctFilename = '!nonexistent';
-
-      for (let i = 0; i < filenames.length; i += 1) {
-        if (filenames[i].toLowerCase() === filename.toLowerCase()) {
-          correctFilename = `\`${filenames[i]}\`.`;
-          break;
+  that.getFilenamesInDir(dir, (filenamesPromise) => {
+    filenamesPromise.then(filenames => {
```

Comment:
Hidding whitespace changes is recommended, there is only indent change inside this code block.

Reply:
Resolved: outdated.